app/models/spree/product.rb

```diff
@@ -236,7 +232,10 @@ def properties_including_inherited
       ps = product_properties.all
 
       if inherits_properties
-        ps = OpenFoodNetwork::PropertyMerge.merge(ps, supplier.producer_properties)
+        # NOTE: Set the supplier as the first variant supplier. If variants have different supplier,
+        # result might not be correct
+        supplier = variants.first.supplier
+        ps = OpenFoodNetwork::PropertyMerge.merge(ps, supplier&.producer_properties || [])
```

Comment:
Good to note the caveat. I wonder if this feature is even needed, or if we can remove it.

Reply:
It's used in `Api::ProductSerializer` , which is in turn used in `ProductRendere` here : https://github.com/openfoodfoundation/openfoodnetwork/blob/2676891322ee4ff7e95a6f8a1ac6f17fdf03a213/app/services/products_renderer.rb#L21-L26

This is used to display the products in the producer shop, which include product properties to allow the "Filter by" to work. Ultimately it should be reworked to use the variant instead of the product, but for now it allows us to keep the existing functionality/